frontend/src/pages/projects/types.ts

```diff
@@ -16,10 +16,20 @@ export type UpdateObjectAtPropAndValue<T> = (propKey: keyof T, propValue: ValueO
 
 export type NameDescType = {
   name: string;
-  k8sName?: string;
+  k8sName?: {
+    value: string;
+    isUserInputk8sName?: boolean;
+    isTruncated?: boolean;
+  };
   description: string;
 };
 
+export type K8sNameValidationState = {
+  ruleLength: 'indeterminate' | 'error' | 'success' | 'warning';
+  ruleCharacterTypes: 'indeterminate' | 'error' | 'success';
+  [key: string]: 'indeterminate' | 'error' | 'success' | 'warning';
+};
```

Comment:
Two things:
1. Does this need to be open-ended? I'm not sure why `[key: string]` is needed here
2. Can you use the PF enum as the value instead of copying the string values over? Might make it easier to maintain and use that way (see other comments)

Reply:
@andrewballantyne sorry, 
1. It's a glitch, It was initially defined for different purpose before which was to handle any number of additional properties with keys of type string and values of type valueType. 
2. I have updated with PF enum